runtime/js/10_dispatch_buffer.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+
+  function assert(cond) {
+    if (!cond) {
+      throw Error("assert");
+    }
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////// General async handling //////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////////////////////
+
+  // General Async response handling
+  let nextRequestId = 1;
+  const promiseTable = {};
+
+  function opAsync(opName, opRequestBuilder, opResultParser) {
+    // Make sure requests of this type are handled by the asyncHandler
+    // The asyncHandler's role is to call the "promiseTable[requestId]" function
+    core.setAsyncHandlerByName(opName, (bufUi8, _) => {
+      const [requestId, result, error] = opResultParser(bufUi8, true);
```

Comment:
So we used array destructuring heavily in previous iterations of `core.js` and that caused a lot of GC triggers. Let's see what's the impact on performance...

Reply:
ok, good to know; I guess a workaround can be used to replace this in case performance suffers.

Reply:
@bartlomieju maybe a hacky solution could be to reuse some shared variables and reassign them when returning?

Reply:
No, let's not do that, it would be harder to reason about. Let's see if there will be a negative perf impact and if so address in a follow up